src/main/java/org/opensearch/knn/index/engine/faiss/AbstractFaissMethod.java

```diff
@@ -105,6 +106,19 @@ static KNNLibraryIndexingContext adjustPrefix(
         if (knnMethodConfigContext.getVectorDataType() == VectorDataType.BINARY) {
             prefix = FAISS_BINARY_INDEX_DESCRIPTION_PREFIX;
         }
+        if (knnMethodConfigContext.getVectorDataType() == VectorDataType.BYTE) {
+
+            // If VectorDataType is Byte using Faiss engine then manipulate Index Description to use "SQ8_direct_signed" scalar quantizer
+            // For example, Index Description "HNSW16,Flat" will be updated as "HNSW16,SQ8_direct_signed"
+            String indexDescription = methodAsMapBuilder.indexDescription;
+            if (indexDescription != null && indexDescription.isEmpty() == false) {
```

Comment:
why not just use StringUtils.isEmpty() from apache commons.

Also what will happen in the case indexDescription is null or empty. I think we should throw an exception rather than just silently not doing anything.

Reply:
Just a heads up - as part of the mode/compression support, Im going to adjust the default logic to provide kinda of a default provider as opposed to just a default. So we wont need this additional logic once thats changed, because we will be able to supply default quantizer based on data type.

Reply:
Updated to use StringUtils for now. But, I'm not throwing an exception for the else case as Jack mentioned that we will be refactoring it.